01-messaging.md

```diff
@@ -476,6 +476,75 @@ every message maximally).
 Finally, the usage of periodic `ping` messages serves to promote frequent key
 rotations as specified within [BOLT #8](08-transport.md).
 
+### Peer storage
+
+## The `peer_storage` and `your_peer_storage` Messages
+
+Nodes that advertise the `option_provide_storage` feature offer storing
+arbitrary data for their peers. The data stored must not exceed 65531 bytes,
+which lets it fit in lightning messages.
+
+Nodes that advertise the `option_want_storage` feature want to have their
+data stored by their peers that support `option_provide_storage`.
+
+Nodes can verify that their `option_provide_storage` peers correctly store
+their data at each reconnection, by comparing the contents of the
+retrieved data with the last one they sent. However, nodes should not expect
+their peers to always have their latest data available.
+
+Nodes ask their peers to store data using the `peer_storage` message & expect peers to return them latest data using `your_peer_storage` message:
+
+1. type: 7 (`peer_storage`)
+2. data:
+   * [`u16`: `length`]
+   * [`length*byte`:`blob`]
```

Comment:
Could also extend this in order to have key-values instead of one single storage blob?
```suggestion
   * [`u16`:`key`]
   * [`length*byte`:`blob`]
```

Reply:
Yes, good idea, but I think that would be more useful in case we want to distribute multiple backups (or backup with >65kb data).

In that case, we'd also have to specify which `key` would be considered `ack`, and this could lead to spamming (i.e. Peer can send me many blobs that could fill up my storage) that is why the current scheme just replaces the old storage blob with a new one.

Reply:
> Could also extend this in order to have key-values instead of one single storage blob?

Under the hood it might be a key-value structure - but think that that is better left to the above layer.

Reply:
> Under the hood it might be a key-value structure - but think that that is better left to the above layer.

+1. Single Opaque Buffer is simplest to implement and all other data structures can be implemented atop this if we want. KV storage also introduces the question of what we are supposed to send back on reconnect.

Reply:
Yeah, not a great idea to go kv